### PR TITLE
Make compatible with other JVM Locales as for example turkish.

### PR DESCRIPTION
**Pull request by janitza-bjag**
capitalize() without Locale.ENGLISH leads to unexpected chars which are language specific

## Checklist

- [x] Pull requests follows the [contribution guide](https://github.com/graphql-java-kickstart/graphql-java-tools/wiki/Contribution-guide)
- [x] New or modified functionality is covered by tests

### CODE REVIEW

**Comment by florensie** [comment]
I'd maybe use `Locale.ROOT` instead.

**Comment by janitza-bjag** [comment]
> I'd maybe use `Locale.ROOT` instead.

Seems to be the "better" approach, done

File: src/main/kotlin/graphql/kickstart/tools/resolver/FieldResolverScanner.kt [hunk 1]
```diff
@@ -14,6 +14,7 @@ import org.slf4j.LoggerFactory
 import java.lang.reflect.Method
 import java.lang.reflect.Modifier
 import java.lang.reflect.Type
+import java.util.*
```
Comment (florensie) [inline src/main/kotlin/graphql/kickstart/tools/resolver/FieldResolverScanner.kt:17]: Don't do wildcard imports.

File: src/main/kotlin/graphql/kickstart/tools/util/Utils.kt [hunk 2]
```diff
@@ -8,6 +8,7 @@ import kotlinx.coroutines.Dispatchers
 import java.lang.reflect.Method
 import java.lang.reflect.ParameterizedType
 import java.lang.reflect.Proxy
+import java.util.*
```
Comment (florensie) [inline src/main/kotlin/graphql/kickstart/tools/util/Utils.kt:11]: Same thing

File: src/test/kotlin/graphql/kickstart/tools/FieldResolverScannerTest.kt [hunk 3]
```diff
@@ -97,6 +107,10 @@ class FieldResolverScannerTest {
         fun getHullType(): HullType = HullType()
     }
 
+    class capitalizeQuery : GraphQLQueryResolver {
```
Comment (florensie) [inline src/test/kotlin/graphql/kickstart/tools/FieldResolverScannerTest.kt:110]: Classes should be pascal case.

**Review by florensie — CHANGES_REQUESTED** [review]

File: src/main/kotlin/graphql/kickstart/tools/resolver/FieldResolverScanner.kt (hunk 1 above)
Comment (janitza-bjag) [reply src/main/kotlin/graphql/kickstart/tools/resolver/FieldResolverScanner.kt]: done

File: src/main/kotlin/graphql/kickstart/tools/util/Utils.kt (hunk 2 above)
Comment (janitza-bjag) [reply src/main/kotlin/graphql/kickstart/tools/util/Utils.kt]: done

File: src/test/kotlin/graphql/kickstart/tools/FieldResolverScannerTest.kt (hunk 3 above)
Comment (janitza-bjag) [reply src/test/kotlin/graphql/kickstart/tools/FieldResolverScannerTest.kt]: done

**Review by florensie — APPROVED** [review]

File: src/main/kotlin/graphql/kickstart/tools/resolver/FieldResolverScanner.kt (hunk 1 above)
Comment (florensie) [reply src/main/kotlin/graphql/kickstart/tools/resolver/FieldResolverScanner.kt]: Feel free to mark my suggestions as resolved, doesn't seem like I can do this.

**Review by janitza-bjag — COMMENTED** [review]
use Locale.ROOT to avoid language specific capitalize effects

**Review by vojtapol — APPROVED** [review]
Thank you for your contribution.

**Comment by janitza-bjag** [comment]
Is there anything to do for me?

**Comment by oryan-block** [comment]
Looks like `capitalize` is deprecated now: https://youtrack.jetbrains.com/issue/KT-43023
We should reconsider this approach.

**Comment by oryan-block** [comment]
@janitza-bjag lets replace these with `"x".replaceFirstChar(Char::titlecase)` please.

**Review by janitza-bjag — COMMENTED** [review]
recommended changes applied

File: src/test/kotlin/graphql/kickstart/tools/FieldResolverScannerTest.kt [hunk 4]
```diff
@@ -81,6 +81,16 @@ class FieldResolverScannerTest {
         assertEquals((meta as MethodFieldResolver).method.returnType, HullType::class.java)
     }
 
+    @Test
+    fun `scanner finds field resolver method using capitalize field_name`() {
+        val resolver = RootResolverInfo(listOf(CapitalizeQuery()), options)
+
+        val meta = scanner.findFieldResolver(FieldDefinition("id", TypeName("HullType")), resolver)
```
Comment (oryan-block) [inline src/test/kotlin/graphql/kickstart/tools/FieldResolverScannerTest.kt:88]: Lets make sure the test will fail on all machines by setting the default locale to Turkish `Locale.forLanguageTag("tr-TR")` and resetting it back to default at the end. 
And add a comment to explain why we do that please.

**Review by oryan-block — CHANGES_REQUESTED** [review]

**Comment by oryan-block** [comment]
Closing this since we're unable to push/merge it as it is. Opened #565 with the same commits.

**Pull request closed without merging**